### PR TITLE
Add SoilAndPeatland categories and SiteType categories

### LINKED ISSUE #15: SoilAndPeatlandCategory enumerations
From MelaSoilAndPeatlandCategory , work out the necessary Enum classes for VMI and Forest Centre.
Create the internal SoilAndPeatlandCategory Enum as a combination of VMI and Forest Centre values.
Work as with https://github.com/menu-hanke/forest-data-model/issues/4.

Refer to vmi-data-converter with the current VMI -> Mela and Forest Centre -> Mela transformation logic.
If transformation is direct (requires no computation or multiple sources), add transformation maps into the relevant forestdatamodel.conversion files.
If transformation requires computation or multiple sources, create a new issue for the transformation work.

Browse https://extra.bitcomp.fi/metsastandardi_ehdotus/V20/MV/doc_01/index29.html#id155 for reference with the Forest Centre category data. See VMI manual PDFs for VMI reference.

### PR DESCRIPTION
**Pull request by unie31**
Closes #15

### CODE REVIEW

File: forestdatamodel/conversion/internal2mela.py [hunk 1]
```diff
@@ -78,13 +77,66 @@
...
+def soil_peatland_mapper(target):
+    """determining the soil or peatland type requires knowing the site type (fertility type).
+    Make sure to set it first, because if it's not set for the target object, this method will raise an exception.
+    """
+
+    #UNSPECIFIED_TREELESS_MIRE (only exists for VMI data) can be either rich (letto) or barren (neva).
+    #Thus, the MELA soil/peatland type can be deduced with the information about the site type:
+    if target.soil_peatland_category == SoilPeatlandCategory.UNSPECIFIED_TREELESS_MIRE:
+        if target.site_type_category is None:
+            raise TypeError
```
Comment (unie31) [inline forestdatamodel/conversion/internal2mela.py:120]: With this I tried to capture the bug found in the earlier implementation where soil_peatland_category was determined wrongly when site_Type_category was not set. However I have a feeling that raising may not be the ideal solution so I'm open to improvement suggestions.

Comment (jpy-luke) [reply forestdatamodel/conversion/internal2mela.py:120]: Rather return None

File: forestdatamodel/enums/internal.py [hunk 2]
```diff
@@ -69,3 +69,23 @@ class OwnerCategory(Enum):
...
     OTHER_COMMUNITY = 9
     UNDIVIDED = 10
+
+class SoilPeatlandCategory(IntEnum):
+    MINERAL_SOIL = 1 # kangas
+    SPRUCE_MIRE = 2 # korpi
+    PINE_MIRE = 3 # räme
+    BARREN_TREELESS_MIRE = 4 # neva
+    RICH_TREELESS_MIRE = 5 # letto
+    UNSPECIFIED_TREELESS_MIRE = 6 # VMI "avosuo"
```
Comment (jpy-luke) [inline forestdatamodel/enums/internal.py:79]: Didn't we agree that just TREELESS_MIRE is enough? Specifying barren or rich is done via the SiteType specifically upon a need for it, like with Mela conversion.

**Review by jpy-luke — CHANGES_REQUESTED** [review]
See line comments

File: forestdatamodel/conversion/internal2mela.py [hunk 3]
```diff
@@ -109,22 +108,20 @@ def site_type_mapper(target):
...
-    #Thus, the MELA soil/peatland type can be deduced with the information about the site type:
-    if target.soil_peatland_category == SoilPeatlandCategory.UNSPECIFIED_TREELESS_MIRE:
+    if target.soil_peatland_category == SoilPeatlandCategory.TREELESS_MIRE:
         if target.site_type_category is None:
-            raise TypeError 
-        #note that the comparision needs to be done against mela site type because it is converted to it above.
+            target.soil_peatland_category = None
+            return target
+
         if target.site_type_category in __mela_rich_mire_types:
```
Comment (jpy-luke) [inline forestdatamodel/conversion/internal2mela.py:120]: No need for early return. Early returns generally make code more difficult to understand with multiple exit points from functions. Make the second if an `elif`. This allows us to trust at a glance that the if-else construct of this function always does only one thing.

**Review by jpy-luke — CHANGES_REQUESTED** [review]
Another comment to resolve

**Pull request merged**